Well completion unit tests

avanti-em opened the pull request:
Basic unit tests for Well Completion-settings component

hkfb commented:
Please merge latest upstream and make sure that the checks are passing. You can run `npm run validate` locally to run the linting and typescript checks locally.

aprilzhang commented:
Thanks @avanti-em for working on this PR 👍 
I have rebased this PR onto the latest master and fixed the errors related to importing SmartNodeSelector modules. There are a couple of tests needed to be updated (I marked TODO in the code).
One thing to note is that the line import "jest-styled-components"; in .test.jsx is quite essential to make the snapshot test work properly.

avanti-em commented:
> Thanks @avanti-em for working on this PR 👍
> I have rebased this PR onto the latest master and fixed the errors related to importing SmartNodeSelector modules. There are a couple of tests needed to be updated (I marked TODO in the code).
> One thing to note is that the line import "jest-styled-components"; in .test.jsx is quite essential to make the snapshot test work properly.

Thank you so much @aprilzhang for solving all those issues. I will start working on the TODOs. 
Last point noted 👍

aprilzhang commented on src/lib/components/WellCompletions/components/Settings/FilterMenu.tsx:
```diff
@@ -22,6 +22,7 @@ const FilterMenu: React.FC = React.memo(() => {
         <div>
             <Tooltip title={isDrawerOpen ? "Close filter menu" : "Filter"}>
                 <Button
+                    data-testid="filter_button"
```
Avanti has also pointed out this new attribute can be removed using https://www.npmjs.com/package/babel-plugin-react-remove-properties. Maybe something to be refined in the future.
